src/Paket.Core/PaketConfigFiles/ProjectFile.fs

```diff
@@ -1180,11 +1180,20 @@ module ProjectFile =
 
 
     let save forceTouch project =
+        let hasUtf8Bom fileName =
+            if File.Exists fileName then
+                use fs = new FileStream(fileName, FileMode.Open, FileAccess.Read)
+                let bomBuffer = Array.zeroCreate 4
+                fs.Read(bomBuffer, 0, 4) |> ignore
```

Comment:
Then the next line is most likely false or undefined.

Comparing the result with 4 would definitely make future readers more happy ;)